File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int right = 0, left = 0, deleted = 0, maxL = 0;

        while (right < nums.size()) {
            if (nums[right] == 0) deleted++;

            if (deleted > 1) {
                if (nums[left] == 0) deleted--;
                left++;
            }

            maxL = max(maxL, right - left);

            right++;
        }

        return maxL;
    } //runtime : O(N) , space : O(1)
};

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        start, maxWindow, lastZero = 0,0,-1
        n = len(nums)

        for i in range(n):
            if nums[i] == 0:
                start = lastZero + 1
                lastZero = i
            maxWindow = max(maxWindow, i - start)

        return maxWindow
                
            